### PR TITLE
Run Mac hostonly tests on any available arch

### PR DESCRIPTION
**Pull request by jmagman**
Allow Mac hostonly (not devicelab or benchmark) tests to run on any available architecture, not just x64 Intel machines.

Example expands available bots for this category of tests from
https://chromium-swarm.appspot.com/botlist?c=id&c=task&c=cpu&c=os&c=status&d=asc&f=pool%3Aluci.flutter.prod&f=os%3AMac-12&f=cpu%3Ax86&f=device_type%3Anone&s=id
to
https://chromium-swarm.appspot.com/botlist?c=id&c=task&c=cpu&c=os&c=status&d=asc&f=pool%3Aluci.flutter.prod&f=os%3AMac-12&f=device_type%3Anone&s=id

Also run `tool_host_cross_arch_tests` on both archs, as that's the intention of that shard.

Remove `arm64ruby` dependency, unnecessary as of https://github.com/flutter/flutter/issues/110561.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: .ci.yaml
Comment (jmagman) [inline .ci.yaml:68]: First, attempting to run these tests only on arm64 to confirm they all pass on that arch.

Comment (jmagman) [reply .ci.yaml:68]: (This change will not be checked in, just confirming on this PR that all tasks pass on arm64 as well as x64).

**Comment by jmagman** [comment]
@keyonghan what else needs to be done to make sure the arm64 cipd packages are used on arm bots, and Rosetta is installed?

https://ci.chromium.org/ui/p/flutter/builders/try/Mac%20module_test_ios/17563/overview
```
Target aot_assembly_release failed: ProcessException: Bad CPU type in executable
```

https://ci.chromium.org/ui/p/flutter/builders/try/Mac%20build_tests_1_4/26563/overview
```
  failed to resolve flutter_internal/browsers/chrome/mac-arm64@version:98.1 (line 1): no such package: flutter_internal/browsers/chrome/mac-arm64
```
Ideally this behavior is tied to the cpu property of the bot and not the name of the task/builder so that it can run on any bot with any architecture, given available capacity.

**Comment by keyonghan** [comment]
For Rosetta, filed https://bugs.chromium.org/p/chromium/issues/detail?id=1354817
For arm64 cipd packages, we need the package maintainer to add the arm64 version, for example `chrome`.

**Comment by jmagman** [comment]
> For arm64 cipd packages, we need the package maintainer to add the arm64 version, for example `chrome`.

This one failed missing `openjdk` but we should have that? https://github.com/flutter/flutter/issues/98532
https://ci.chromium.org/ui/p/flutter/builders/try/Mac%20framework_tests_misc/25668/overview

**Comment by jmagman** [comment]
And this one is missing arm ruby
https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8805363977658226209/+/u/run_test.dart_for_tool_host_cross_arch_tests_shard_and_subshard_None/test_stdout

https://chrome-infra-packages.appspot.com/p/flutter/ruby/mac-arm64/+/P1b-yT8l6N4J6TbVI5fJDqmbWIKP6WhKcLfITjWZ2ZEC
https://github.com/flutter/flutter/issues/87508#issuecomment-1076614727

**Comment by keyonghan** [comment]
> This one failed missing openjdk but we should have that? https://github.com/flutter/flutter/issues/98532
https://ci.chromium.org/ui/p/flutter/builders/try/Mac%20framework_tests_misc/25668/overview

https://github.com/flutter/flutter/pull/110110 to fix.

**Comment by jmagman** [comment]
> For arm64 cipd packages, we need the package maintainer to add the arm64 version, for example `chrome`.

Filed https://github.com/flutter/flutter/issues/110113

**Comment by jmagman** [comment]
@keyonghan everything passed when I added the `arm64` tag.  If we want `Mac` to work for either, we can't set it in .ci.yaml, it should be derived from the `cpu` property.  Where is that tag used?

**Comment by jmagman** [comment]
> @keyonghan everything passed when I added the `arm64` tag. If we want `Mac` to work for either, we can't set it in .ci.yaml, it should be derived from the `cpu` property. Where is that tag used?

Wait, that tag didn't even get applied? I guess the tags on the individual tasks override it, not add to it?
```
          "tags": [
                "devicelab",
                "hostonly"
              ],
```
https://chromium-swarm.appspot.com/task?id=5ce567548c6b2a10&o=true&w=true

**Comment by jmagman** [comment]
Hm the passing tests actually ran on x86 even though they had the `cpu` `arm64` property:
https://chromium-swarm.appspot.com/task?id=5ce5675befae1110&o=true&w=true

```
cpu:x86x86-64x86-64-i7-8700B
```
```
        "requested_properties": {
...
            "cpu": "arm64",
```

**Comment by keyonghan** [comment]
> @keyonghan everything passed when I added the `arm64` tag. If we want `Mac` to work for either, we can't set it in .ci.yaml, it should be derived from the `cpu` property. Where is that tag used?

The tag is used only for data analysis purpose, and it should not affect platform selection.
Sounds something wrong when fetching properties/dimensions from PR to overriding the scheduled build config, though the first commit did pick up the changes in the PR.

**Comment by jmagman** [comment]
Blocked on getting Rosetta on these machines https://bugs.chromium.org/p/chromium/issues/detail?id=1354817

**Comment by jmagman** [comment]
> Hm the passing tests actually ran on x86 even though they had the `cpu` `arm64` property

This happened again...

**Comment by jmagman** [comment]
Filed https://github.com/flutter/flutter/issues/110561 to install arm Ruby on arm bots.

File: .ci.yaml
Comment (jmagman) [inline .ci.yaml:79]: First, attempting to run these tests only on arm64 to confirm they all pass on that arch.
(This change will not be checked in, just confirming on this PR that all tasks pass on arm64 as well as x64).

**Review by keyonghan — APPROVED** [review]
LGTM!

Shall we note somewhere (maybe in the wiki) that for any new Mac host only target, the owner should validate it works on both arches?

**Comment by jmagman** [comment]
> Shall we note somewhere (maybe in the wiki) that for any new Mac host only target, the owner should validate it works on both arches?

Which wiki were you thinking? Maybe here? go/flutter-luci-builders#sdk-sharded-tests

I think once https://github.com/flutter/flutter/issues/110561 and https://github.com/flutter/flutter/issues/110113 are done _most_ tests won't have any arch-specific needs.

**Comment by keyonghan** [comment]
> Which wiki were you thinking? Maybe here? go/flutter-luci-builders#sdk-sharded-tests

I am thinking about public docs like CI_YAML readme: https://github.com/flutter/cocoon/blob/main/CI_YAML.md, or https://github.com/flutter/flutter/wiki/Adding-a-new-Test-Shard#steps-to-add-a-new-framework-test-shard.

**Comment by jmagman** [comment]
I'm going to block this until https://github.com/flutter/flutter/issues/110561 is done, it's causing too many failures on arm, and it's too possible for devs adding new tests to have it pass in presubmit on x64, but fail flakily post-submit on arm.

**Comment by jmagman** [comment]
Trying all arm tests again on top of https://github.com/flutter/flutter/issues/110561.

**Comment by jmagman** [comment]
This is ready for review now.  The x64 tests remaining are related to the missing arm Chrome cipd package, and the web team is looking into that: https://github.com/flutter/flutter/issues/110113

File: .ci.yaml
Comment (jmagman) [reply .ci.yaml:2865]: That's this PR, what should it be a link to?

Comment (keyonghan) [reply .ci.yaml:2865]: I was expecting a bug about why this is marked as flaky, but then i realized this is a "new" target. The link LGTM.

File: .ci.yaml
Comment (jmagman) [reply .ci.yaml:2719]: Something different, fix at https://github.com/flutter/flutter/pull/112034

Comment (keyonghan) [reply .ci.yaml:2719]: Could you add a link here?

File: .ci.yaml
Comment (keyonghan) [inline .ci.yaml:2904]: The `runtime` can be dropped now.

File: .ci.yaml
Comment (jmagman) [reply .ci.yaml]: Was trying to quickly fix this so I could run as arm but I couldn't get it done in time.  Link added!

File: .ci.yaml
Comment (jmagman) [reply .ci.yaml]: Done.

**Review by keyonghan — APPROVED** [review]
LGTM!

**Pull request merged**
